httpcore5/src/main/java/org/apache/hc/core5/net/URIBuilder.java

```diff
@@ -264,13 +264,39 @@ public URIBuilder setScheme(final String scheme) {
     }
 
     /**
-     *Sets the URI scheme specific part.
+     * Sets the URI scheme specific part.
      *
      * @param schemeSpecificPart
-     * @param nvps Optional, can be null. List of query parameters to be reused by the specific scheme
      * @return this.
+     * @since 5.1
      */
-    public URIBuilder setSchemeSpecificPart(final String schemeSpecificPart, final List <NameValuePair> nvps) {
+    public URIBuilder setSchemeSpecificPart(final String schemeSpecificPart) {
+        this.encodedSchemeSpecificPart = schemeSpecificPart;
+        return this;
+    }
+
+    /**
+     * Sets the URI scheme specific part and append a list of parameters to this part.
+     *
+     * @param schemeSpecificPart
+     * @param nvps Optional, can be null. List of NameValuePair query parameters to be reused by the specific scheme part
+     * @return this.
+     * @since 5.1
+     */
+    public URIBuilder setSchemeSpecificPartWithParameter(final String schemeSpecificPart, final NameValuePair... nvps) {
```

Comment:
@netsrotr `setSchemeSpecificPart` was a better name if you ask me. Why `setSchemeSpecificPartWithParameter`?

Reply:
The overloads taking `NameValuePair...` and `List<NameValuePair>` cannot have the same name.
So I voted for an explicit setter (that match the getter) and differentiate the other by `...Parameter(...` and `...Parameters(...` 
But: I can take whatever you prefer... Suggestions?

Reply:
> The overloads taking NameValuePair... and List<NameValuePair> cannot have the same name
@netsrotr Are you sure? We already have `setPathSegments` overloaded methods.

Reply:
Oops... ;-) you are right: the unit test points me to the wrong road... Will refine the PR. Sry